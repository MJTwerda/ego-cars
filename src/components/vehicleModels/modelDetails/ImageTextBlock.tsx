import { stripHtmlTags } from '@/utils/stringsManagement';
import { Typography } from '@mui/material';
import Image from 'next/image';
import { ImageBlock, TextInfoBlock } from '../interfaces/textBlock.interfaces';

interface ImageTextBlockProps {
  image: ImageBlock;
  textInfo: TextInfoBlock;
};

const ImageTextBlock = ({ image, textInfo }: ImageTextBlockProps) => {

  return (
    <div className={`flex flex-col lg:flex-row ${image.position === 'right' ? 'lg:flex-row-reverse' : ''} align-center mx-4 mt-11 sm:mx-32`}>
      <Image
        src={image.src}
        alt={image.alt}
        width={300}
        height={300}
        className={`rounded ${image.width ? `lg:w-[${image.width}%]` : 'lg:w-1/2'} ${image.position === 'right' ? 'lg:ml-12' : 'lg:mr-12'}`}
      />

      <div className="flex flex-col self-center lg:w-[40%]">
        {textInfo.model &&<Typography variant="h5">{textInfo.model}</Typography>}
        <Typography variant={textInfo.title.variant} className="mb-4">{textInfo.title.text}</Typography>
        <Typography variant="body2">
          {`${stripHtmlTags(textInfo.description)} Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.`}
        </Typography>
      </div>
    </div>
  );
};

export default ImageTextBlock;