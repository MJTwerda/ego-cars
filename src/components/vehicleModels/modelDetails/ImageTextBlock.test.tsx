import { act, render, RenderResult, waitFor } from '@testing-library/react';
import ImageTextBlock, { ImageTextBlockProps } from './ImageTextBlock';

const mockPropsWhitoutModel: ImageTextBlockProps = {
  image: {
    src: '/img1.jpg',
    alt: 'Image 1',
    position: 'left'
  },
  textInfo: {
    title: {
      text: 'Title 1',
      variant: 'h4'
    },
    description: 'Description 1'
  }
}

describe('Testing <ImageTextBlock /> component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Snapshot component whitout "model" prop', async()=> {
    let component: RenderResult;

    await act(async () => {
      component = render(<ImageTextBlock image={mockPropsWhitoutModel.image} textInfo={mockPropsWhitoutModel.textInfo} />);
    });

    await waitFor(() => {
      expect(component).toMatchSnapshot();

      const images = component.getAllByRole('img');
      expect(images.length).toBe(1);
      expect(images[0]).toHaveAttribute('alt', 'Image 1');
      expect(images[0]).toHaveClass('lg:mr-12');

      const titleInfo = component.getByText('Title 1');
      expect(titleInfo).toBeInTheDocument();
    });
  });

  test('Snapshot component with "model" prop', async()=> {
    const mockPropsWhitModel: ImageTextBlockProps = {
      ...mockPropsWhitoutModel,
      image: {
        ...mockPropsWhitoutModel.image,
        position: 'right'
      },
      textInfo: {
        ...mockPropsWhitoutModel.textInfo,
        description: '<p>Description 2</p>',
        model: 'Model 1'
      }
    };
    let component: RenderResult;

    await act(async () => {
      component = render(<ImageTextBlock image={mockPropsWhitModel.image} textInfo={mockPropsWhitModel.textInfo} />);
    });

    await waitFor(() => {
      expect(component).toMatchSnapshot();
      const image = component.getByRole('img');
      expect(image).toBeInTheDocument();
      expect(image).toHaveAttribute('alt', 'Image 1');
      expect(image).toHaveClass('lg:ml-12');
      const modelInfo = component.getByText('Model 1');
      expect(modelInfo).toBeInTheDocument();
    });

  });
});