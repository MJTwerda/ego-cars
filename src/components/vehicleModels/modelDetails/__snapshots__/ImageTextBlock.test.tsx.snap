// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ImageTextBlock /> component Snapshot component whitout "model" prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col lg:flex-row  align-center mx-4 mt-11 sm:mx-32"
      >
        <img
          alt="Image 1"
          class="rounded lg:w-1/2 lg:mr-12"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
        <div
          class="flex flex-col self-center lg:w-[40%]"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 mb-4 css-f84ov1-MuiTypography-root"
          >
            Title 1
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
          >
            Description 1 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col lg:flex-row  align-center mx-4 mt-11 sm:mx-32"
    >
      <img
        alt="Image 1"
        class="rounded lg:w-1/2 lg:mr-12"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="flex flex-col self-center lg:w-[40%]"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 mb-4 css-f84ov1-MuiTypography-root"
        >
          Title 1
        </h4>
        <p
          class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
        >
          Description 1 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing <ImageTextBlock /> component Snapshot component with "model" prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col lg:flex-row lg:flex-row-reverse align-center mx-4 mt-11 sm:mx-32"
      >
        <img
          alt="Image 1"
          class="rounded lg:w-1/2 lg:ml-12"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
        <div
          class="flex flex-col self-center lg:w-[40%]"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
          >
            Model 1
          </h5>
          <h4
            class="MuiTypography-root MuiTypography-h4 mb-4 css-f84ov1-MuiTypography-root"
          >
            Title 1
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
          >
            Description 2 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col lg:flex-row lg:flex-row-reverse align-center mx-4 mt-11 sm:mx-32"
    >
      <img
        alt="Image 1"
        class="rounded lg:w-1/2 lg:ml-12"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="flex flex-col self-center lg:w-[40%]"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
        >
          Model 1
        </h5>
        <h4
          class="MuiTypography-root MuiTypography-h4 mb-4 css-f84ov1-MuiTypography-root"
        >
          Title 1
        </h4>
        <p
          class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
        >
          Description 2 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
